src/formatDate.js

```diff
@@ -8,7 +8,58 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separator);
+
+  const dateCompare = {};
+
+  fromFormat.forEach((part, index) => {
+    switch (part) {
+      case 'YYYY':
+        dateCompare['YYYY'] = dateParts[index];
+        break;
+      case 'YY':
+        dateCompare['YY'] = dateParts[index];
+        break;
+      case 'MM':
+        dateCompare['MM'] = dateParts[index];
+        break;
+      case 'DD':
+        dateCompare['DD'] = dateParts[index];
+        break;
+      default:
+        break;
+    }
```

Comment:
The 'dateCompare' object is used to map the parts of the date to their corresponding format. The use of an object with keys corresponding to date parts ('YYYY', 'YY', 'MM', 'DD') is a good approach to keep track of the date components. However, consider using a more descriptive name for this object to reflect its purpose, such as 'dateComponents' or 'datePartsMap'.